modules/experimental-layers/src/tile-layer/tile-layer.js

```diff
@@ -0,0 +1,50 @@
+import {GeoJsonLayer, CompositeLayer} from 'deck.gl';
+import TileCache from './utils/tile-cache';
+
+const defaultProps = {
+  renderSubLayers: props => new GeoJsonLayer(props)
+};
+
+export default class TileLayer extends CompositeLayer {
+  initializeState() {
+    this.state = {
+      tiles: [],
+      tileCache: new TileCache({fetchData: this.props.fetchData})
+    };
+  }
+
+  shouldUpdateState({changeFlags}) {
+    return changeFlags.somethingChanged;
+  }
+
+  updateState({props, oldProps, context, changeFlags}) {
+    if (props.fetchData !== oldProps.fetchData) {
```

Comment:
We usually avoid directly comparing two props incase inline functions are used, e.g.:

```js
new ScatterplotLayer({
  ...
  getRadius: d => d.radius
})
```

Then `props.getRadius !== oldProps.getRadius` would always be true.

Instead, deck.gl requires explicit [updateTriggers](http://deck.gl/#/documentation/deckgl-api-reference/layers/layer?section=data-properties) to trigger data regeneration. You can access the comparison results via:

```js
changeFlags.updateTriggersChanged &&
        (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged.fetchData))
```

Reply:
Makes sense. 

Is `changeFlags.updateTriggersChanged.all` checking whether all accessor props are changed?

Reply:
The user can have an updateTrigger called `all` which would invalidate all accessors.